crates/telemetry/src/config.rs

```diff
@@ -0,0 +1,78 @@
+use envconfig::Envconfig;
+use opentelemetry::KeyValue;
+use opentelemetry_semantic_conventions::resource::SERVICE_VERSION;
+
+/// Provides configuration for the telemetry system.
+///
+/// Largely consists of OpenTelemetry variables defined
+/// [here](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#general-sdk-configuration)
+/// but also has some Spin specific configuration.
+#[derive(Envconfig, Debug)]
+pub struct Config {
+    /// Disables OpenTelemetry for all signals (traces and metrics).
+    #[envconfig(from = "OTEL_SDK_DISABLED", default = "false")]
+    pub otel_sdk_disabled: bool,
+
+    /// Value of the `service.name` resource attribute.
+    #[envconfig(from = "OTEL_SERVICE_NAME", default = "spin")]
+    pub otel_service_name: String,
+
+    /// Key-value pairs to be used as resource attributes.
+    #[envconfig(from = "OTEL_RESOURCE_ATTRIBUTES", default = "")]
+    pub otel_resource_attributes: KeyValues,
+
+    /// Default exporter endpoint for both traces and metrics.
+    #[envconfig(from = "OTEL_EXPORTER_OTLP_ENDPOINT")]
+    pub otel_exporter_otlp_endpoint: Option<String>,
+
+    /// Exporter endpoint for traces.
+    #[envconfig(from = "OTEL_EXPORTER_OTLP_TRACES_ENDPOINT")]
+    pub otel_exporter_otlp_traces_endpoint: Option<String>,
+
+    /// Exporter endpoint for metrics.
+    #[envconfig(from = "OTEL_EXPORTER_OTLP_METRICS_ENDPOINT")]
+    pub otel_exporter_otlp_metrics_endpoint: Option<String>,
```

Comment:
We don't technically use these config values other than to see if the user has set them so that we know if we should turn on an OTEL layer or not. `tracing-otlp` automatically reads these values from the environment.

Reply:
It might be less confusing to just list the vars that can enable tracing e.g.:

```rust
fn otlp_enabled() -> bool {
    const ENABLING_VARS: &[&str] = &[
        "OTEL_EXPORTER_OTLP_ENDPOINT",
        // ...
    ];
    ENABLING_VARS.iter().any(|key|
        std::env::var_os(key).is_some_and(|val| !val.is_empty()))
}
```

Reply:
Yup, I like that more.